tests/common/fixtures/consistency_checker/consistency_checker.py

```diff
@@ -0,0 +1,335 @@
+import pytest
+import logging
+import json
+import os
+import datetime
+from collections import defaultdict
+
+logger = logging.getLogger(__name__)
+
+SYNCD_CONTAINER = "syncd"
+QUERY_ASIC_SCRIPT = "query-asic.py"
+LIBSAIREDIS_DEB = "libsairedis.deb"
+PYTHON3_PYSAIREDIS_DEB = "python3-pysairedis.deb"
+DUT_DST_PATH_HOST = "/tmp/consistency-checker"
+DUT_DST_PATH_CONTAINER = "/consistency-checker"
+
+DUT_SCRIPT_PATH_SRC = os.path.dirname(__file__) + "/" + QUERY_ASIC_SCRIPT
+DUT_SCRIPT_PATH_DST_HOST = DUT_DST_PATH_HOST + "/" + QUERY_ASIC_SCRIPT
+DUT_SCRIPT_PATH_DST_CONTAINER = DUT_DST_PATH_CONTAINER + "/" + QUERY_ASIC_SCRIPT
+
+
+class ConsistencyChecker:
+
+    def __init__(self, duthost, libsairedis_download_url=None, python3_pysairedis_download_url=None):
+        """
+        If the libsairedis_download_url and python3_pysairedis_download_url are provided, then these artifacts
+        are downloaded and installed on the DUT, otherwise it's assumed that the environment is already setup
+        for the consistency checker.
+        """
+        self._duthost = duthost
+        self._libsairedis_download_url = libsairedis_download_url
+        self._python3_pysairedis_download_url = python3_pysairedis_download_url
+
+    def __enter__(self):
+        logger.info("Initializing consistency checker on dut...")
+
+        self._duthost.shell(f"mkdir -p {DUT_DST_PATH_HOST}")
+        self._duthost.copy(src=DUT_SCRIPT_PATH_SRC, dest=DUT_SCRIPT_PATH_DST_HOST)
+
+        if self._libsairedis_download_url is not None:
+            self._duthost.shell(f"curl -o {DUT_DST_PATH_HOST}/{LIBSAIREDIS_DEB} {self._libsairedis_download_url}")
+        if self._python3_pysairedis_download_url is not None:
+            self._duthost.shell(
+                f"curl -o {DUT_DST_PATH_HOST}/{PYTHON3_PYSAIREDIS_DEB} {self._python3_pysairedis_download_url}")
+
+        # Move everything into syncd container
+        self._duthost.shell((
+            f"docker cp {DUT_DST_PATH_HOST} {SYNCD_CONTAINER}:/ && "
+            f"rm -rf {DUT_DST_PATH_HOST}"
+        ))
+
+        if self._python3_pysairedis_download_url is not None:
+            # Install python3-sairedis in syncd container
+            self._duthost.shell((f"docker exec {SYNCD_CONTAINER} bash -c "
```

Comment:
DEB installation failures are not caught and not logged.

Reply:
The kwarg module_ignore_errors kwarg is implicitly False here. So if the deb installation fails an error will be raised with the failing command. I don't think we should catch and log and continue.